### PR TITLE
Reduce the scope of some critical sections in the examples

### PR DESCRIPTION
**Pull request by torkeldanielsson**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -35,7 +35,7 @@ cast = { version = "0.2", default-features = false }
 cortex-m = "0.6"
 cortex-m-rt = "0.6"
 embedded-hal = { version = "0.2", features = ["unproven"] }
-stm32f0 = "0.10"
+stm32f0 = "0.11"
```
Comment (therealprof) [inline Cargo.toml:38]: Please do this in a separate PR, we also need to add a CHANGELOG entry for it since that is a breaking change.

Comment (torkeldanielsson) [reply Cargo.toml]: Ok!

**Comment by therealprof** [comment]
bors try
